packages/nextra-theme-docs/src/components/collapse.tsx

```diff
@@ -1,86 +1,52 @@
 import { useRef, useEffect, ReactElement, ReactNode } from 'react'
-import cn from 'clsx'
+import { useMounted } from 'nextra/hooks'
 
 export function Collapse({
   children,
   className,
-  open
+  isOpen,
+  horizontal = false
 }: {
   children: ReactNode
   className?: string
-  open: boolean
+  isOpen: boolean
+  horizontal?: boolean
 }): ReactElement {
   const containerRef = useRef<HTMLDivElement>(null)
   const innerRef = useRef<HTMLDivElement>(null)
-  const animationRef = useRef<any>()
-  const initialRender = useRef(true)
-  const initialState = useRef(open)
+  const initialOpen = useRef(isOpen)
+  const mounted = useMounted()
 
   useEffect(() => {
-    if (initialRender.current) return
+    const container = containerRef.current
+    const inner = innerRef.current
+    if (!mounted || !container || !inner) return
 
-    if (animationRef.current) {
-      clearTimeout(animationRef.current)
-    }
-    if (open) {
-      const container = containerRef.current
-      const inner = innerRef.current
-      if (container && inner) {
-        const contentHeight = innerRef.current.clientHeight
-        container.style.maxHeight = `${contentHeight}px`
-        container.classList.remove('nx-duration-500')
-        container.classList.add('nx-duration-300')
+    container.classList.toggle('nx-duration-500', !isOpen)
+    container.classList.toggle('nx-duration-300', isOpen)
 
-        inner.style.opacity = '1'
-        animationRef.current = setTimeout(() => {
-          const container = containerRef.current
-          if (container) {
-            // should be style property in kebab-case, not css class name
-            container.style.removeProperty('max-height')
-          }
-        }, 300)
-      }
+    if (horizontal) {
+      // save initial width to avoid word wrapping when container width will be changed
+      inner.style.width = `${inner.clientWidth}px`
+      container.style.width = `${isOpen ? inner.clientWidth : 0}px`
```

Comment:
I added support horizontal collapsing and simplified a lot this component

## initial open

https://user-images.githubusercontent.com/7361780/212500951-4be801c8-7300-44c5-8862-14a3b02fd8b4.mov

## initial closed

https://user-images.githubusercontent.com/7361780/212500955-4461041d-c900-4b72-897c-0197cd7db3ba.mov

Reply:
Nice!